src/osu_allgatherv.jl

```diff
@@ -0,0 +1,42 @@
+export OSUAllgatherv
+
+struct OSUAllgatherv <: MPIBenchmark
+    conf::Configuration
+    name::String
+end
+
+function OSUAllgatherv(T::Type=Float32;
+                   filename::Union{String,Nothing}="julia_osu_allgatherv.csv",
+                   kwargs...,
+                   )
+    return OSUAllgatherv(
+        Configuration(T; filename, max_size=2 ^ 20, kwargs...),
+        "OSU ALLgatherv",
+    )
+end
+
+function osu_allgatherv(T::Type, bufsize::Int, iters::Int, comm::MPI.Comm)
+    rank = MPI.Comm_rank(comm)
+    nranks = MPI.Comm_size(comm)
+    send_buffer = ones(T, bufsize)
+    recv_buffer = zeros(T, bufsize * nranks)
+    counts = [bufsize for _ in 1:nranks]
```

Comment:
This is slightly simpler with [`fill`](https://docs.julialang.org/en/v1/base/arrays/#Base.fill) (and it's ok to use `fill` because `bufsize::Int` is immutable)
```suggestion
    counts = fill(bufsize, nranks)
```